### PR TITLE
Incluster CA

### PR DESCRIPTION
**Pull request by maykulkarni**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1 [hunk 1]
```diff
@@ -0,0 +1,889 @@
+$INFO_IMP = "Yellow"
+$INFO = "Cyan"
+$ERR = "Red"
+$SUCC = "Green"
+
+function InvokeRestAPICall($EndPoint, $Method, $Body, $ErrorMessage)
+{
```
Comment (v-zhgup) [inline src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1]: Use class-method model to implement the commands

File: src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1 [hunk 2]
```diff
@@ -0,0 +1,889 @@
+$INFO_IMP = "Yellow"
+$INFO = "Cyan"
+$ERR = "Red"
```
Comment (v-zhgup) [inline src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1]: We have PublishCustomMessage method and MessageType enum for printing message of different types like 'Warning/Error/Default' etc

File: src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1 [hunk 3]
```diff
@@ -0,0 +1,889 @@
...
+        Write-Host $ErrorMessage -ForegroundColor $ERR
+        return $null
+    }
+    return $response
+}
+
+
+function Insert-StuffIntoDB($SecretScopeName, $SecretKeyName, $Secret) {
+    $params = @{
+      'scope' = $SecretScopeName;
```
Comment (v-zhgup) [inline src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1]: Use appropriate names for methods

File: src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1 [hunk 4]
```diff
@@ -0,0 +1,889 @@
...
+    }
+    # $response = $dbresource
+    # $rg_name = $response.ResourceGroupName
+    $PAT = Read-Host -Prompt 'Personal Access Token(PAT)'
+    $IK = Read-Host -Prompt "Input Instrumentation Key for enabling App Insights (press enter to skip)"
+    $PersonalAccessToken = $PAT.Trim()
+    $InstrumentationKey = $IK.Trim()
+    $WorkSpaceBaseUrl = "https://" +  $response.Location + ".azuredatabricks.net"
+    # Please don't modify these values
+    $ConfigBaseUrl = "https://azsdkossep.azureedge.net/3.9.0/"
```
Comment (v-zhgup) [inline src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1]: Use variables used similar to what is used in RunbookCoreSetup.ps1 to update the cdn value from build pipeline

File: src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1 [hunk 5]
```diff
@@ -0,0 +1,889 @@
...
+     {
+        $SecretScopeAlreadyExists = $true
+        Write-Host "Secret scope [$SecretScopeName] already exists in the workspace. We will reuse it." -ForegroundColor $INFO_IMP
+     }
+    }
+    
+    # Create Secret Scope if not already exists
+    if(-not $SecretScopeAlreadyExists)
+    {
+        Write-Host "Creating a new secret scope [$SecretScopeName] in the workspace" -ForegroundColor $INFO
```
Comment (v-zhgup) [inline src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1]: use $this.PublishCustomMessage to print messages on terminal and powershell output file, Write-Host will only write on terminal

File: src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1 [hunk 6]
```diff
@@ -0,0 +1,895 @@
...
+    {
+        # Prepare job schedule
+        $Schedule = '"0 0 #h# * * ?"'
+        $jobHrs = ((Get-Date).ToUniversalTime().Hour + 1) % 24
+        $Schedule = $Schedule -replace '\#h\#', $jobHrs
+
+        # Create job
+        Write-Host "Creating Job 'AzSK_CA_Scan_Job' in the workspace" -ForegroundColor $INFO
+        $JobConfigServerUrl = $ConfigBaseUrl + "DatabricksCAScanJobConfig.json"
+        #$JobConfigServerUrl = "C:\Users\makul\Desktop\ADB_Config.json"
```
Comment (v-zhgup) [inline src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1]: Remove commented code display your local path

File: src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1 [hunk 7]
```diff
@@ -0,0 +1,895 @@
...
+        Write-Host "For more info, please see docs: https://aka.ms/devopskit/inclusterca" -ForegroundColor $INFO
+    }
+  
+    #end region
+
+}
+
+
+function Update-DataBricks() {
+    [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.SecurityProtocolType]::Tls12;
```
Comment (v-zhgup) [inline src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1]: Is updating TLS protocol on user's machine by our code not impact other stuff, should not we put some warning/note if it necessary to run this script

Comment (maykulkarni) [reply src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1]: Without that we can't execute API calls. It is limited to only the current PS session and will not affect the environment.

File: src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1 (hunk 1 above)
Comment (maykulkarni) [reply src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1]: Will be using class methods in next sprint

File: src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1 (hunk 2 above)
Comment (maykulkarni) [reply src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1]: Will be using class methods in next sprint

File: src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1 (hunk 3 above)
Comment (maykulkarni) [reply src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1]: Completed

File: src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1 (hunk 5 above)
Comment (maykulkarni) [reply src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1]: This is not completely integrated with the main module so it won't get anything from the constructor

File: src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1 (hunk 6 above)
Comment (maykulkarni) [reply src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1]: Completed

File: src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1 (hunk 4 above)
Comment (v-zhgup) [reply src/AzSK/InClusterCA/ContinuousAssuranceForClusters.ps1]: its fine you can create another variable in that, we can do that in next sprint

**Pull request merged**